flow-client/src/main/java/com/vaadin/client/flow/util/ClientJsonCodec.java

```
@@ -85,18 +95,37 @@ public static StateNode decodeStateNode(StateTree tree, JsonValue json) {
      */
     public static Object decodeWithTypeInfo(StateTree tree, JsonValue json) {
         if (json.getType() == JsonType.OBJECT) {
-            // Check for @v object format
-            String type = getStringProperty(json, "@v");
-            if ("node".equals(type)) {
-                int nodeId = (int) getNumberProperty(json, "id");
+            // Check for compact @v-node format
+            JsonValue nodeIdValue = getProperty(json, "@v-node");
+            if (nodeIdValue != null && nodeIdValue.getType() == JsonType.NUMBER) {
+                int nodeId = (int) nodeIdValue.asNumber();
                 Node domNode = tree.getNode(nodeId).getDomNode();
                 return domNode;
-            } else if ("return".equals(type)) {
-                int nodeId = (int) getNumberProperty(json, "nodeId");
-                int channelId = (int) getNumberProperty(json, "channelId");
-                return createReturnChannelCallback(nodeId, channelId,
-                        tree.getRegistry().getServerConnector());
             }
+            
+            // Check for compact @v-return format
+            JsonValue returnArray = getProperty(json, "@v-return");
+            if (returnArray != null && returnArray.getType() == JsonType.ARRAY) {
```

Not sure if its relevant but it might be more resilient to raise an error rather than just ignoring if the value is not an array with the right length